app/server/src/features/accounts/methods.unit.test.ts

```diff
@@ -0,0 +1,418 @@
+                confirmNewPassword: newPassword,
+            };
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+            // Assert
+            await expect(AccountMethods.updatePassword(prismaMock, input)).rejects.toThrow(
+                'New password must be at least 8 characters.'
+            );
+        });
+        test('should throw if password does not contain at least 1 number and 1 symbol', async () => {
+            // Arrange
+            const user = { ...userData };
+            const newPassword = 'invalidPassword';
+            const input = {
+                email: userData.email,
+                oldPassword: userTextPassword,
+                newPassword,
+                confirmNewPassword: newPassword,
+            };
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+            // Assert
+            await expect(AccountMethods.updatePassword(prismaMock, input)).rejects.toThrow(
+                'New password must contain a mixture of lowercase and uppercase letters, at least one number, and at least one special character.'
+            );
+        });
+        test('should throw if new passwords do not match', async () => {
+            // Arrange
+            const user = { ...userData };
+            const newPassword = 'newPassword1!';
+            const input = {
+                email: userData.email,
+                oldPassword: userTextPassword,
+                newPassword,
+                confirmNewPassword: 'mismatchedNewPassword',
+            };
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+            // Assert
+            await expect(AccountMethods.updatePassword(prismaMock, input)).rejects.toThrow('Passwords must match.');
+        });
+        test('should update password sucessfully', async () => {
+            // Arrange
+            const user = { ...userData };
+            const newPassword = 'newPassword1!';
+            const newPasswordHashed = await bcrypt.hash(newPassword, 10);
+            const updatedUser = { ...userData, password: newPasswordHashed };
+            const input = {
+                email: userData.email,
+                oldPassword: userTextPassword,
+                newPassword,
+                confirmNewPassword: newPassword,
+            };
+            prismaMock.user.findUnique.mockResolvedValueOnce(user);
+            prismaMock.user.update.mockResolvedValueOnce(updatedUser);
+
+            // Act
+            const output = await AccountMethods.updatePassword(prismaMock, input);
+
+            // Assert
+            const userWithGlobalId = toUserId(updatedUser);
+            const token = await jwt.sign({ id: userWithGlobalId.id });
+            const expectedOutput = { updatedUser, token };
+            expect(output).toEqual(expectedOutput);
+        });
+    });
+
+    describe('deleteAccount', () => {
+        test('should throw if account has no existing password', async () => {
```

Comment:
No changes needed. This just seems odd in a policy sense that we can't delete accounts with no password set.

Reply:
Yeah it is quite odd, I didn't really think about it when I was just going through testing all the logic but I think a few of the account methods might need some tweaking. I'll do that along with the OWASP changes.